internal/blocktx/store/postgresql/postgres_test.go

```diff
@@ -474,6 +651,24 @@ func TestPostgresDB(t *testing.T) {
 		// then
 		assert.Equal(t, expectedUnverifiedBlockHeights, res.UnverifiedBlockHeights)
 	})
+
+	t.Run("lock blocks table", func(t *testing.T) {
+		err := postgresDB.WriteLockBlocksTable(context.Background())
+		require.Error(t, err)
+		require.Equal(t, ErrNoTransaction, err)
+
+		uow, err := postgresDB.StartUnitOfWork(context.Background())
+		require.NoError(t, err)
+
+		err = uow.WriteLockBlocksTable(context.Background())
+		require.NoError(t, err)
+
+		err = uow.Rollback()
+		require.NoError(t, err)
+
+		err = uow.Commit()
+		require.Equal(t, ErrNoTransaction, err)
+	})
```

Comment:
Here we could have some additional checks to ensure the correct behaviour of commits/rollbacks etc

```go
t.Run("lock blocks table", func(t *testing.T) {
		prepareDb(t, postgresDB, "")

		err := postgresDB.WriteLockBlocksTable(context.Background())
		require.Error(t, err)
		require.Equal(t, ErrNoTransaction, err)

		uow, err := postgresDB.StartUnitOfWork(context.Background())
		require.NoError(t, err)

		err = uow.WriteLockBlocksTable(context.Background())
		require.NoError(t, err)

		err = uow.Rollback()
		require.NoError(t, err)

		err = uow.Commit()
		require.Equal(t, ErrNoTransaction, err)

		// ensure writing is possible after commit
		blockHash := testutils.RevChainhash(t, "000000000000000001b8adefc1eb98896c80e30e517b9e2655f1f929d9958a48")
		expectedBlock := &blocktx_api.Block{
			Hash:   blockHash[:],
			Height: 100,
			Status: blocktx_api.Status_LONGEST,
		}

		_, err = postgresDB.UpsertBlock(ctx, expectedBlock)
		require.NoError(t, err)

		// ensure commit persists a change
		blockHash2 := testutils.RevChainhash(t, "0000000000000000064cbaac5cedf71a5447771573ba585501952c023873817b")
		expectedBlock2 := &blocktx_api.Block{
			Hash:   blockHash2[:],
			Height: 101,
			Status: blocktx_api.Status_LONGEST,
		}
		uow, err = postgresDB.StartUnitOfWork(context.Background())
		require.NoError(t, err)

		err = uow.WriteLockBlocksTable(context.Background())
		require.NoError(t, err)

		_, err = uow.UpsertBlock(ctx, expectedBlock2)
		require.NoError(t, err)

		err = uow.Commit()
		require.NoError(t, err)

		block, err := postgresDB.GetBlock(ctx, blockHash)
		require.NoError(t, err)
		require.Equal(t, expectedBlock.Hash, block.Hash)

		// ensure rollback does roll back a transaction
		uow, err = postgresDB.StartUnitOfWork(context.Background())
		require.NoError(t, err)

		err = uow.WriteLockBlocksTable(context.Background())
		require.NoError(t, err)

		blockHash3 := testutils.RevChainhash(t, "0000000000000000025855b62f4c2e3732dad363a6f2ead94e4657ef96877067")
		expectedBlock3 := &blocktx_api.Block{
			Hash:   blockHash3[:],
			Height: 102,
			Status: blocktx_api.Status_LONGEST,
		}

		_, err = uow.UpsertBlock(ctx, expectedBlock3)
		require.NoError(t, err)

		err = uow.Rollback()
		require.NoError(t, err)

		_, err = postgresDB.GetBlock(ctx, blockHash3)
		require.ErrorIs(t, err, store.ErrBlockNotFound)
	})
```

Reply:
I also tried to ensure writing to the locked table fails, but that is blocking infinitely.

Tried to check if it is possible to ensure `UpsertBlock` is not blocking in that case but returning an error, but haven't found out how.

Reply:
> if it is possible to ensure UpsertBlock is not blocking in that case but returning an error

Yes, it's absolutely possible, but I made this blocking on purpose, so that if 2 reorgs would happen at the same  time (trying to acquire the same lock), one would wait for the other to finish and then execute. I didn't want it to throw an error.

If you want it to throw an error when 2 instances try to access the lock:
`LOCK TABLE blocktx IN SHARE MODE`

And with blocking / no error:
`LOCK TABLE blocktx IN EXCLUSIVE MODE`


Anyways, the locking mechanism is now removed, as after going through all the edge cases, it turned out it's not necessary. So I removed it completely.